src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +784,59 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend
+ */
+void
+AssignGlobalPID(void)
+{
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		uint64 globalPID = GenerateGlobalPID();
+
+		SpinLockAcquire(&MyBackendData->mutex);
+		MyBackendData->globalPID = globalPID;
+		SpinLockRelease(&MyBackendData->mutex);
+
+		return;
+	}
+
+	StringInfo applicationName = makeStringInfo();
+	appendStringInfoString(applicationName, application_name);
+
+	char *globalPIDString = &applicationName->data[6];
```

Comment:
Directly accessing to `[6]` seems unusual. Can we use something else, like `strstr` function or such? A very quick idea I can think of:

```C

if (strstr(application_name, CITUS_APP_NAME_PREFIX))
char *globalPIDString = &applicationName->data[strnlen(CITUS_APP_NAME_PREFIX)];
```

Reply:
I'll add `CITUS_APP_NAME_PREFIX` but I already check if the `applicationName` starts with `citus-` in the `IsCitusInitiatedRemoteBackend`. Is that really necessary?

Reply:
Well, I think it is not _strictly_ necessary, but accessing `[6]` is:
(a) hard to understand what is going on / what we are doing
(b) could be hard to debug/understand if we one day decide to change  `CITUS_APP_NAME_PREFIX`
(c) has a dependency to `IsCitusInitiatedRemoteBackend` call, which may get refactored

So, it is mostly a readability and some robustness to refactors issue from my perspective

Reply:
related, can we have an ExtractGlobalPID function to maintain the string formatting logic in a single place?